src/main/java/uk/gov/justice/hmpps/prison/service/MovementsService.java

```diff
@@ -320,9 +317,8 @@ private void logErrorAndThrowBadRequest(final String errorMessage) {
         final var datesToTry = splitDatesIfTheySpanAcrossDifferentDays(fromDateTime, toDateTime);
 
         return datesToTry.stream()
-            .flatMap(date -> movementsRepository.getIndividualSchedules(date).stream())
+            .flatMap(date -> movementsRepository.getIndividualSchedules(agencyIds, date).stream())
             .filter(isTransferAndNotDeleted())
```

Comment:
isn't this filter an "obvious" one to move to database? not sure what other sort events there are but maybe transfer ones are a small minority?